backend/tests/data_layer/data_saver/test_sqlite_saver.py

```
@@ -53,128 +214,671 @@ def sqlite_config() -> DictConfig:
-    inserted_data = [info.model_dump() for info in stock_price_info]
 
-    expected_data = [InstrumentPrice(**data).to_dict() for data in kafka_data]
+def test_database_connection_error() -> None:
+    """
+    Test handling of database connection errors.
+    """
+    mock_consumer = MagicMock()
 
-    assert inserted_data == expected_data
+    # Use invalid database path that would cause connection issues
+    with pytest.raises(Exception):  # Could be various database-related exceptions
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)